include/ddc/kernels/fft.hpp

```diff
@@ -326,15 +372,42 @@ struct kwArgs_core
 };
 
 // N,a,b from x_mesh
+/**
+ * @brief Get the mesh size along a given dimension.
+ *
+ * @tparam DDim The dimension along which the mesh size is returned.
+ * @param x_mesh The mesh.
+ *
+ * @return The mesh size along the required dimension.
+ */
 template <typename DDim, typename... DDimX>
 int N(ddc::DiscreteDomain<DDimX...> x_mesh)
 {
     static_assert(
             (is_uniform_point_sampling_v<DDimX> && ...),
             "DDimX dimensions should derive from UniformPointSampling");
-    return ddc::get<DDim>(x_mesh.extents());
+    return static_cast<int>(x_mesh.template extent<DDim>());
 }
 
+/**
+ * @brief Get the lower boundary coordinate along a given dimension.
+ *
+ * The lower boundary of the spatial domain (which appears in Nyquist-Shannon theorem) is not
+ * xmin=ddc::coordinate(x_mesh.front()). Indeed, this coordinate identifies the lower cell, but
+ * the lower boundary is the left side of this lower cell, which is a = xmin - cell_size/2, with
+ * cell_size = (b-a)/N. It leads to a = xmin-(b-a)/2N. The same derivation for the
+ * upper boundary coordinate gives b = xmax+(b-a)/2N. Inverting this linear system leads to:
+ *
+ * a = ((2N-1)*xmin-xmax)/2/(N-1)
+ * b = ((2N-1)*xmax-xmin)/2/(N-1)
+ *
+ * The current function implements the first equation.
```

Comment:
I don't understand what you mean by "coordinate of the lower cell". Let's avoid mentioning cells for now.

Reply:
```
| * | * | * | * | * |
| |                 | |
a xmin           b
                   xmax
```